airbyte-integrations/connectors/source-microsoft-sharepoint/source_microsoft_sharepoint/stream_reader.py

```diff
@@ -197,7 +197,15 @@ def drives(self):
         """
         Retrieves and caches SharePoint drives, including the user's drive based on authentication type.
         """
-        drives = execute_query_with_retry(self.one_drive_client.drives.get())
+        drives = None
```

Comment:
Why not `drives = list()` instead? Then you can just append to the list below, instead of having the if/else flow.

Reply:
didn't want to change the type of "drives" variable, as it would have a much bigger impact. At the moment it already combines multiple drives.

Reply:
@natikgadzhi have you got examples of such integration tests? I don't see any such test in this connecter. Above change has been tested on a real multi-site sharepoint account
yes, a new parameter is a good idea, unfortunately, right now don't have capacity.